contract/LibSubmarine.sol

```diff
@@ -43,135 +36,177 @@ contract LibSubmarine is ProvethVerifier {
         uint8 _result,
         address indexed _sender,
         address indexed _submarine,
-        uint256 _slashAmount,
+        uint256 _slashAmount/*,
         bytes _proofBlob,
-        bytes _unsignedCommitTx
+        bytes _unsignedCommitTx*/ // Removed due to stack limit
     );
     event Finalized(
         bytes32 indexed _sessionId,
         address indexed _sender,
         uint256 _amountTemp,
         uint256 _revealDeposit,
-        bytes _dataTemp
+        bytes _commitData
     );
 
+    /////////////
+    // Storage //
+    /////////////
 
-    constructor(uint256 _revealDepositAmount, uint32 _challengePeriodLength) public {
-        revealDepositAmount = _revealDepositAmount;
-        challengePeriodLength = _challengePeriodLength;
+    uint256 public revealDepositAmount;
+    uint32 public challengePeriodLength;
+    uint8 public vee = 27;
+
+    mapping(bytes32 => SessionData) public sessionData;
+
+    struct SessionData {
+        bool unlocked;        // set in unlock, used in reveal and finalize
```

Comment:
`revealBlock` and `commitValue` cannot be in the hash because it is used in the sanity checks for the unlock tx. If the sanity checks can be removed we would save about 30k gas. Maybe this can be done with #25?